include/exadg/time_integration/time_int_bdf_base.cpp

```diff
@@ -18,360 +18,52 @@
  *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
  *  ______________________________________________________________________
  */
-
 #include <exadg/time_integration/time_int_bdf_base.h>
 
 namespace ExaDG
 {
-template<typename Number>
-TimeIntBDFBase<Number>::TimeIntBDFBase(double const        start_time_,
-                                       double const        end_time_,
-                                       unsigned int const  max_number_of_time_steps_,
-                                       unsigned int const  order_,
-                                       bool const          start_with_low_order_,
-                                       bool const          adaptive_time_stepping_,
-                                       RestartData const & restart_data_,
-                                       MPI_Comm const &    mpi_comm_,
-                                       bool const          is_test_)
-  : TimeIntBase(start_time_,
-                end_time_,
-                max_number_of_time_steps_,
-                restart_data_,
-                mpi_comm_,
-                is_test_),
-    order(order_),
+TimeIntBDFBase::TimeIntBDFBase(double const        start_time_,
+                               double const        end_time_,
+                               unsigned int const  max_number_of_time_steps_,
+                               unsigned const      order_,
+                               bool const          start_with_low_order_,
+                               bool const          adaptive_time_stepping_,
+                               RestartData const & restart_data_,
+                               MPI_Comm const &    mpi_comm_,
+                               bool const          is_test_)
+  : TimeIntMultistepBase(start_time_,
+                         end_time_,
+                         max_number_of_time_steps_,
+                         order_,
+                         start_with_low_order_,
+                         adaptive_time_stepping_,
+                         restart_data_,
+                         mpi_comm_,
+                         is_test_),
     bdf(order_, start_with_low_order_),
-    extra(order_, start_with_low_order_),
-    start_with_low_order(start_with_low_order_),
-    adaptive_time_stepping(adaptive_time_stepping_),
-    time_steps(order_, -1.0)
-{
-}
-
-template<typename Number>
-void
-TimeIntBDFBase<Number>::setup(bool const do_restart)
```

Comment:
it would be good if all this code would not show up as code changes. Essentially, it looks like a renaming `TimeIntBDFBase` -> `TimeIntMultistepBase` took place. Then, some functionality has been extracted from `TimeIntMultistepBase` into a new class `TimeIntBDFBase`. It would be better if some names are non-sense for now and are changed in an upcoming PR for the sake of showing in this PR on the code that has been changed. I mean that `TimeIntBDFBase` means -- for the time being -- a base class for multistep time integration, and you introduce in this PR another class `TimeIntBDF : public TimeIntBDFBase`.

Reply:
Can we instead have one PR with two seperate commits?